crd/crd.go

```diff
@@ -56,6 +61,11 @@ func NewDatabaseCRD() *apiextv1.CustomResourceDefinition {
 											MaxLength:   intptr(16),
 											Pattern:     DBUsernamePattern,
 										},
+										"password": {
+											Type:                   "object",
+											Description:            "Password to access the database",
+											XPreserveUnknownFields: boolptr(true),
```

Comment:
used the `x-kubernetes-preserve-unknown-fields` than spending time to build the right validation rule for this field.

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#rawextension

Reply:
note that otherwise field pruning happens. see https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#controlling-pruning